polyfill/lib/absolute.mjs

```diff
@@ -1,126 +1,197 @@
-});
+  static fromEpochSeconds(epochSecondsParam) {
+    const epochMilliseconds = ES.ToNumber(epochSecondsParam) * 1000;
+    const resultObject = Object.create(Absolute.prototype);
+    CreateSlots(resultObject);
+    SetSlot(resultObject, EPOCHNANOSECONDS, { ms: epochMilliseconds, ns: 0 });
+    return resultObject;
+  }
+  static fromEpochMilliseconds(epochMillisecondsParam) {
+    const epochMilliseconds = ES.ToNumber(epochMillisecondsParam);
+    const resultObject = Object.create(Absolute.prototype);
+    CreateSlots(resultObject);
+    SetSlot(resultObject, EPOCHNANOSECONDS, { ms: epochMilliseconds, ns: 0 });
+    return resultObject;
+  }
+  static fromEpochMicroseconds(epochMicroseconds) {
+    if ('bigint' !== typeof epochNanoseconds) throw RangeError('bigint required');
+    const epochMilliseconds = epochMicroseconds / BigInt(1e3);
+    const restNanoseconds = epochMicroseconds % BigInt(1e3);
+    const resultObject = Object.create(Absolute.prototype);
+    CreateSlots(resultObject);
+    SetSlot(resultObject, EPOCHNANOSECONDS, { ms: epochMilliseconds, ns: restNanoseconds });
+    return resultObject;
+  }
+  static fromEpochNanoseconds(epochNanoseconds) {
+    if ('bigint' !== typeof epochNanoseconds) throw RangeError('bigint required');
+    const epochMilliseconds = epochNanoseconds / BigInt(1e6);
+    const restNanoseconds = epochNanoseconds % BigInt(1e6);
+    const resultObject = Object.create(Absolute.prototype);
+    CreateSlots(resultObject);
+    SetSlot(resultObject, EPOCHNANOSECONDS, { ms: epochMilliseconds, ns: restNanoseconds });
+    return resultObject;
+  }
+  static fromString(isoString) {
+    isoString = ES.ToString(isoString);
+    const match = STRING.exec(isoString);
+    if (!match) throw new RangeError(`invalid absolute: ${isoString}`);
+    const year = ES.ToInteger(match[1]);
+    const month = ES.ToInteger(match[2]);
+    const day = ES.ToInteger(match[3]);
+    const hour = ES.ToInteger(match[4]);
+    const minute = ES.ToInteger(match[5]);
+    const second = ES.ToInteger(match[6]);
+    const millisecond = ES.ToInteger(match[7]);
+    const microsecond = ES.ToInteger(match[8]);
+    const nanosecond = ES.ToInteger(match[9]);
+    const zone = match[11] || match[10] || 'UTC';
+    const datetime = ES.GetIntrinsic('%Temporal.datetime%', {
+      year,
+      month,
+      day,
+      hour,
+      minute,
+      second,
+      millisecond,
+      microsecond,
+      nanosecond
+    });
+    return datetime.inZone(zone || 'UTC', match[11] ? match[10] : 'earlier');
+  }
+  static from(...args) {
+    return ES.GetIntrinsic('%Temporal.absolute%')(...args);
+  }
+  static compare(one, two) {
+    one = ES.GetIntrinsic('%Temporal.absolute%')(one);
+    two = ES.GetIntrinsic('%Temporal.absolute%')(two);
+    one = GetSlot(one, EPOCHNANOSECONDS);
+    two = GetSlot(two, EPOCHNANOSECONDS);
+    if (one.ms !== two.ms) return two.ms - one.ms;
+    if (one.ns !== two.ns) return two.ns - one.ns;
```

Comment:
I think it'd be best to return 0, -1 or 1, rather than these more complex numbers.